src/main/java/seedu/address/model/lead/Lead.java

```diff
@@ -0,0 +1,63 @@
+package seedu.address.model.lead;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents the potential of a client.
+ */
+public class Lead {
+    public static final String MESSAGE_CONSTRAINTS = "Lead should only take values hot|warm|cold";
+    private final LeadType leadType;
+
+    /**
+     * Returns a lead given a string. Checks if the string is a valid lead;
+     *
+     * @param lead the string representation for the lead.
+     */
+    public Lead(String lead) {
+        checkArgument(isValidLead(lead.toUpperCase()), MESSAGE_CONSTRAINTS);
+        this.leadType = Enum.valueOf(LeadType.class, lead.toUpperCase());
+    }
+
+    public Lead(LeadType lead) {
+        this.leadType = lead;
+    }
+
+    /**
+     * Determines whether a given string is a valid lead.
+     * A valid lead is either "hot", "warm" or "cold".
+     *
+     * @param lead the string to check
+     * @return true if the string is a valid lead, false otherwise
+     */
+    public static boolean isValidLead(String lead) {
```

Comment:
For our purposes (since there will only be 3 lead types), this method is okay, but worth noting that this can usually be made more generalised with Enum functionality. (like .name() for each item in enum)